maps_dashboards/public/components/map_container/map_container.tsx

```diff
@@ -81,24 +83,58 @@ export const MapContainer = ({
       }
     }
 
-    if (maplibreRef.current) {
-      maplibreRef.current.on('click', onClickMap);
-      for (const layer of tooltipEnabledLayers) {
-        DocumentLayerFunctions.addTooltip(maplibreRef.current, layer);
+    function onHoverMap(e: MapMouseEvent) {
+      setCoordinates(e.lngLat);
+
+      // remove previous popup
+      hoverPopup?.remove();
+
+      const features = maplibreRef.current?.queryRenderedFeatures(e.point);
+      if (features && maplibreRef.current) {
+        hoverPopup = createPopup({
+          features,
+          layers: tooltipEnabledLayers,
+          showCloseButton: false,
+          showPagination: false,
+          showLayerSelection: false,
+        });
+        hoverPopup
+          ?.setLngLat(getPopupLngLat(features[0].geometry) ?? e.lngLat)
+          .addTo(maplibreRef.current);
       }
     }
 
+    if (maplibreRef.current) {
+      const map = maplibreRef.current;
+      map.on('click', onClickMap);
+      // reset cursor to default when user is no longer hovering over a clickable feature
+      map.on('mouseleave', () => {
+        map.getCanvas().style.cursor = '';
+        hoverPopup?.remove();
+      });
+      map.on('mouseenter', () => {
+        map.getCanvas().style.cursor = 'pointer';
+      });
+      // add tooltip when users mouse move over a point
+      map.on('mousemove', onHoverMap);
+    }
+
     return () => {
       if (maplibreRef.current) {
         maplibreRef.current.off('click', onClickMap);
+        maplibreRef.current.off('mousemove', onHoverMap);
       }
     };
   }, [layers]);
 
   return (
     <div>
       <EuiPanel hasShadow={false} hasBorder={false} color="transparent" className="zoombar">
-        <p> Zoom: {zoom} </p>
+        <small>
+          zoom: {zoom}{' '}
+          {coordinates &&
+            `, lat: ${coordinates.lat.toFixed(4)}, lon: ${coordinates.lng.toFixed(4)}`}
```

Comment:
I just found that this component width may change when lat, lng change, can we make it fixed?

Reply:
updated